modules/squants/shared/src/main/scala/io/circe/squants/package.scala

```diff
@@ -0,0 +1,99 @@
+package io.circe
+
+import _root_.squants._
+import _root_.squants.Quantity
+
+import _root_.squants.experimental.formatter.syntax._
+import _root_.squants.experimental.formatter._
+import _root_.squants.information.{Information, Megabytes}
+import _root_.squants.information.InformationConversions._
+import _root_.squants.experimental.unitgroups.information.{IECInformation, MetricInformation}
+
+import scala.reflect.runtime.universe._
+import scala.util.{Failure, Success, Try}
+
+import scala.collection.Set
+
+
+/**
+  * Provides codecs for [[https://github.com/typelevel/squants squants]] types.
+  *
+  * @author Quentin ADAM @waxzce
+  */
+
+
+package object squants {
+
+  def getCompagnonObject[A <: Quantity[A]](implicit man: TypeTag[A]) = {
+    val universeMirror = runtimeMirror(getClass.getClassLoader)
+    val companionMirror = universeMirror.reflectModule(typeOf[A].typeSymbol.companion.asModule)
+    companionMirror.instance.asInstanceOf[Dimension[A]]
+  }
+
+  implicit def encodeSquantsQuantity[A <: Quantity[A]](implicit formatter: Formatter[A] = null, man: TypeTag[A]): Encoder[A] = new Encoder[A] {
+
+    final def apply(a: A): Json = {
+
+      Json.obj(
+        ("readable", Json.fromString(
+          if (formatter != null) formatter.inBestUnit(a).toString()
+          else {
+            // playing with reflection, prefer to catch exceptions
+            val formatedString: Option[String] = try {
+              import _root_.squants.UnitOfMeasure
+              import _root_.squants.experimental.unitgroups.ImplicitDimensions.space._
+              import _root_.squants.experimental.unitgroups.ImplicitDimensions.information._
+              import _root_.squants.experimental.unitgroups.ImplicitDimensions.electro._
+              import _root_.squants.experimental.unitgroups.ImplicitDimensions.energy._
+              import _root_.squants.experimental.unitgroups.ImplicitDimensions.mass._
+              import _root_.squants.experimental.unitgroups.ImplicitDimensions.motion._
+              import _root_.squants.experimental.unitgroups.ImplicitDimensions.photo._
+              import _root_.squants.experimental.unitgroups.ImplicitDimensions.radio._
+              import _root_.squants.experimental.unitgroups.ImplicitDimensions.thermal._
+              import _root_.squants.experimental.unitgroups.ImplicitDimensions.time._
+              import _root_.squants.experimental.unitgroups.si.strict.implicits._
+              import _root_.squants.experimental.unitgroups.UnitGroup
+              implicit val dim: Dimension[A] = getCompagnonObject[A]
+              val unitGroup:UnitGroup[A] = _root_.squants.experimental.unitgroups.si.strict.implicits.mkSiUnitGroup
+              val defaultformatter = new DefaultFormatter[A](unitGroup)
+              Some(defaultformatter.inBestUnit(a).toString())
+            } catch {
+              case anyException: Throwable => None
+            }
+            formatedString.getOrElse(a.toString())
```

Comment:
This makes me really uncomfortable. I've not used quants much, but in what situations are you going to fall back to `toString`? Personally I'd really rather enumerate instances rather than do this kind of thing.

Reply:
I'll will be switching toString in a rare situations, like, form my actual tests, never. But I prefer to include it here than break.